lambda/matching.ml

```diff
@@ -3205,8 +3208,9 @@ let split_cases tag_lambda_list =
           ((runtime_tag, act) :: consts, nonconsts)
         | Ordinary {runtime_tag}, Variant_boxed _ ->
           (consts, (runtime_tag, act) :: nonconsts)
-        | _, Variant_extensible -> assert false
+        | _, (Variant_extensible | Variant_with_null) -> assert false
```

Comment:
I think the new case here is more "unimplemented" than impossible, but I might be misunderstanding.

Reply:
Oh, right, but it's also resolved in the final PR so not worth editing at this point.